monai/transforms/io/array.py

```
@@ -206,13 +210,24 @@
                 try:
                     self.register(the_reader(*args, **kwargs))
-                except OptionalImportError:
-                    warnings.warn(
-                        f"required package for reader {_r} is not installed, or the version doesn't match requirement."
-                    )
+                except OptionalImportError as e:
+                    msg = f"Required package for reader {_r} is not installed, or the version doesn't match requirement."
+                    if self.raise_on_missing_reader:
+                        raise OptionalImportError(msg) from e
+                    else:
+                        warnings.warn(f"{msg} Will use fallback readers if available.")
```

Ruff (0.12.2)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)